### PR TITLE
bpo-35500: align expected and actual calls on mock.assert_called_with error message.

### PR DESCRIPTION
**Pull request by suhearsawho**
This patch adds an enhancement to the unittest module by adding a newline between the Assertion Error and Expected fields of the error messages generated by mock.assert_called_with. With this addition, the output is clearer as it is easier to read. 

https://bugs.python.org/issue35500

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

File: Misc/NEWS.d/next/Library/2019-02-10-00-00-13.bpo-35500.1HOMmo.rst
Comment (taleinat) [inline Misc/NEWS.d/next/Library/2019-02-10-00-00-13.bpo-35500.1HOMmo.rst]: Thanks for adding a NEWS entry!

This text would be a great description for the PR, but is too verbose for a NEWS entry. A NEWS entry should be a concise description of the change. In this case, I suggest something along the lines of:

```suggestion
Write expected and actual call parameters on separate lines in mock.assert_called_wtih() assertion errors.  Contributed by Susan Su.
```

**Review by taleinat — COMMENTED** [review]
This looks good to me, but I'm still waiting for the issue creator to respond to the suggestions to replace the word "actual" with something longer to achieve vertical alignment between the "Expected" and "Actual" lines.

**Comment by tirkarthi** [comment]
@taleinat @suhearsawho I just responded here : https://bugs.python.org/issue35500#msg335143. I don't really have a strong opinion on the wording and it's a matter of personal preference of mine on using "Expected" and "Actual". It would be a good improvement with the error message on separate lines itself. So I would be happy to see this landing even with a different wording.

**Review by tirkarthi — APPROVED** [review]
LGTM as per https://bugs.python.org/issue35500#msg335145. Please sign the CLA.

Minor nit : You might want to use the below format for hyperlinking to `assert_called_with` method docs.

```rst
Write expected and actual call parameters on separate lines in :meth:`unittest.mock.Mock.assert_called_with` assertion errors.  Contributed by Susan Su.
```

Thanks @suhearsawho for your contribution.

**Review by taleinat — APPROVED** [review]
LGTM. Just waiting for approval of CLA signing.

**Comment by suhearsawho** [comment]
CLA has been approved.

**Comment by suhearsawho** [comment]
Updated the code for the case when self.call_args is None. I declared an actual variable in the mock.py file and testmock.py file to maintain consistency with the rest of the code. Please let me know if there is anything I should revise!

**Comment by lisroach** [comment]
Looks great! Thank you @suhearsawho for your first contribution 😃

**Pull request merged**
